src/main/java/org/elasticsearch/gateway/GatewayAllocator.java

```diff
@@ -284,18 +293,28 @@ public boolean allocateUnassigned(RoutingAllocation allocation) {
             }
 
             if (!canBeAllocatedToAtLeastOneNode) {
+                // still fetching data, remove from the unassigned, and try the next
+                unassignedIterator.remove();
+                routingNodes.ignoredUnassigned().add(shard);
```

Comment:
can we assign ` routingNodes.ignoredUnassigned()` to a variable please? it's used in some places and reads just easier